.github/workflows/generator.yml

```diff
@@ -0,0 +1,44 @@
+name: "Generator"
+
+on:
+  merge_group:
+  pull_request:
+    branches:
+      - "v*.*"
+      - "master"
+      - "feature/*"
+  push:
+    branches:
+      - "v*.*"
+      - "master"
+      - "feature/*"
+
+env:
+  PHP_VERSION: "8.2"
+  # TODO: change to "stable" once 1.20.0 is released
+  # DRIVER_VERSION: "stable"
+  DRIVER_VERSION: "mongodb/mongo-php-driver@v1.20"
+
+jobs:
+  psalm:
+    name: "Diff check"
+    runs-on: "ubuntu-22.04"
+
+    steps:
+      - name: "Checkout"
+        uses: "actions/checkout@v4"
+        with:
+          fetch-depth: 2
```

Comment:
Note that fetch-depth 2 isn't necessary - the default value of 1 to fetch the single commit being tested is sufficient for this action. We use `fetch-depth: 0` in some of the release or merge actions so that we have access to the full history, but since that's not necessary here you can most likely remove this.
```suggestion
```